src/server/components/dashboard/controllers.ts

```diff
@@ -152,12 +157,66 @@ export async function listsController(req: Request, res: Response, next: NextFun
       isNewUser,
       lists,
       csrfToken: getCSRFToken(req),
+      dashboardBoxes: calculateDashboardBoxes(lists as ListsForDashboard[], !req.user?.isAdministrator),
     });
   } catch (error) {
     next(error);
   }
 }
 
+export function calculateSortOrder(
+  queryParamSortOrder: Prisma.ListsForDashboardOrderByWithRelationInput
+): Array<Record<string, string>> {
+  const defaultSortOrder = {
+    country: "asc",
+    type: "asc",
+  };
+
+  // TODO: Object.hasOwn is recommended but is not currently supported by tsc.
+  // eslint-disable-next-line no-prototype-builtins
+  if (!queryParamSortOrder.hasOwnProperty("admins")) {
+    return Object.entries(defaultSortOrder).map(convertEntryToObject);
+  }
+
+  const newSortOrder = {
+    ...queryParamSortOrder,
+    ...defaultSortOrder,
+  };
+
+  return Object.entries(newSortOrder).map(convertEntryToObject);
+}
+
+function convertEntryToObject([key, value]: [string, string]) {
+  return { [key]: value === "desc" ? "desc" : "asc" };
```

Comment:
why can't it just be value? (no ternary?) are you just trying to "validate" someone's passing in asc or desc?

Reply:
correct, I'm just worried someone might enter a random value in the url